### PR TITLE
Improves and fixes for docker-compose

### PR DESCRIPTION
**Pull request by gperdomor**
- Update to use the official swift image as base image.
- Fix for use with docker compose.

Example using docker compose:
```yml
version: "2"
services:
  db:
    image: mysql
    restart: always
    ports:
      - 3306:3306
    environment:
      MYSQL_ROOT_PASSWORD: vapor
      MYSQL_DATABASE: vapor
      MYSQL_USER: vapor
      MYSQL_PASSWORD: vapor

  vapor:
    image: vapor/vapor
    depends_on:
      - db
    ports:
      - 8080:8080
    volumes: 
      - path/to/vapor/directory:/vapor
```

### CODE REVIEW

**Comment by Dog** [comment]
I tried running through your docker file manually since I'm looking to move a project. I got this from it:

```
# curl -sL toolbox.vapor.sh | bash
ℹ️  This script is deprecated.
Please install through Homebrew or APT

📦  Homebrew (macOS):
brew install vapor/tap/vapor

📦  APT (Ubuntu)
eval "$(curl -sL https://apt.vapor.sh)"
sudo apt-get update
sudo apt-get install vapor
```

I then tried to run the apt.vapor.sh directly. I did it with both `curl -sL apt.vapor.sh | bash` and `eval "$(curl -sL https://apt.vapor.sh)"`. They both seem to fail because they try to use sudo.

The apt.vapor.sh file currently is:
```
function add_vapor_apt() {
    eval "$(cat /etc/lsb-release)"

    if [[ "$DISTRIB_CODENAME" != "xenial" && "$DISTRIB_CODENAME" != "yakkety" && "$DISTRIB_CODENAME" != "trusty" ]];
    then
        echo "Only Ubuntu 14.04, 16.04, and 16.10 are supported."
        echo "You are running $DISTRIB_RELEASE ($DISTRIB_CODENAME) [`uname`]"
        return 1;
    fi

    export DEBIAN_FRONTEND=noninteractive

    sudo apt-get -q update
    sudo apt-get -q install -y wget software-properties-common python-software-properties apt-transport-https
    wget -q https://repo.vapor.codes/apt/keyring.gpg -O- | sudo apt-key add -
    echo "deb https://repo.vapor.codes/apt $DISTRIB_CODENAME main" | sudo tee /etc/apt/sources.list.d/vapor.list
    sudo apt-get -q update

    unset DEBIAN_FRONTEND
}

add_vapor_apt;
```

I found I could get it to work if I removed sudo and did:
```
eval "$(cat /etc/lsb-release)"
apt-get -q update
apt-get -q install -y wget software-properties-common python-software-properties apt-transport-https
wget -q https://repo.vapor.codes/apt/keyring.gpg -O- | apt-key add -
echo "deb https://repo.vapor.codes/apt $DISTRIB_CODENAME main" | tee /etc/apt/sources.list.d/vapor.list
apt-get -q update
apt-get install vapor
```

I did notice I could remove wget from the required files if I used curl to grab the key.

```
eval "$(cat /etc/lsb-release)"
apt-get -q update
apt-get -q install -y software-properties-common python-software-properties apt-transport-https
curl -q -o - https://repo.vapor.codes/apt/keyring.gpg | apt-key add -
echo "deb https://repo.vapor.codes/apt $DISTRIB_CODENAME main" | tee /etc/apt/sources.list.d/vapor.list
apt-get -q update
apt-get install vapor
```

**Comment by gperdomor** [comment]
@dog try again please

**Comment by Dog** [comment]
I found I had to change Run to App to get it to build properly for my project. However it still runs into an issue when I try to run it:

```
docker run -it -v $(pwd):/vapor -p 8080:8080 vapor/vapor vapor run
docker: Error response from daemon: driver failed programming external connectivity on endpoint kind_einstein (<removed hash>): Error starting userland proxy: listen tcp 0.0.0.0:8080: bind: address already in use.
```

FWIW I get the same error if I try to use a composer file.

**Comment by gperdomor** [comment]
@Dog running that command will use the official vapor image from master without the modifications of this PR...

**Comment by Dog** [comment]
@gperdomor I removed the official vapor image and built the new one with its tag. It first looks locally before pulling, so it used your build instead of the remote vapor/vapor.

**Comment by gperdomor** [comment]
@Dog The image compiles fine, but i see some problems compiling Vapor 2 projects. This works before, something has changed in the new betas, so i will wait for the stable release before continue...

FYI this is the error trace:
```
Resolving https://github.com/vapor/fluent.git at 2.0.0-beta.15
<unknown>:0: error: unable to build node: '/vapor/.build/release/Random.swiftmodule' (node is produced by multiple commands; e.g., '/vapor/.build/release/CSQLite.build/sqlite3.c.o' and '<Random.module>')
<unknown>:0: error: unable to build node: '/vapor/.build/release/Random.build/URandom.swift.o' (node is produced by multiple commands; e.g., '/vapor/.build/release/CSQLite.build/sqlite3.c.o' and '<Random.module>')
<unknown>:0: error: unable to build node: '/vapor/.build/release/Random.build/RandomProtocol.swift.o' (node is produced by multiple commands; e.g., '/vapor/.build/release/CSQLite.build/sqlite3.c.o' and '<Random.module>')
<unknown>:0: error: unable to build node: '/vapor/.build/release/Random.build/OSRandom.swift.o' (node is produced by multiple commands; e.g., '/vapor/.build/release/CSQLite.build/sqlite3.c.o' and '<Random.module>')
<unknown>:0: error: unable to build node: '/vapor/.build/release/Random.build/Exports.swift.o' (node is produced by multiple commands; e.g., '/vapor/.build/release/CSQLite.build/sqlite3.c.o' and '<Random.module>')
<unknown>:0: error: unable to build node: '/vapor/.build/release/Random.build/EmptyInitializable.swift.o' (node is produced by multiple commands; e.g., '/vapor/.build/release/CSQLite.build/sqlite3.c.o' and '<Random.module>')
<unknown>:0: error: unable to build node: '/vapor/.build/release/Random.build/Array+Random.swift.o' (node is produced by multiple commands; e.g., '/vapor/.build/release/CSQLite.build/sqlite3.c.o' and '<Random.module>')
```

**Comment by gperdomor** [comment]
@Dog working again with latest vapor 2 packages

**Comment by Dog** [comment]
I just got back to this and found that my issue was I had previously installed postgres.app and forgotten about it. Once I uninstalled it from my machine I was able to get through the process without an issue. Thanks

File: vapor/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,11 @@
-FROM vapor/vapor:1.0-xenial
+FROM swift
```
Comment (at-daonguyen) [inline vapor/Dockerfile]: @gperdomor Should we correct Swift version here? `FROM swift:3.1` for example.

Comment (gperdomor) [reply vapor/Dockerfile]: fixed

**Comment by zyxep** [comment]
Merged, and gonna move it to a new branch since Vapor 2 is out.

**Pull request merged**